### PR TITLE
API: add DataTransferItemList

### PR DESCRIPTION
**Pull request by caugner**
[DataTransferItemList](https://developer.mozilla.org/docs/Web/API/DataTransferItemList)

(See also #1763 with the identical `__compat` data.)

### CODE REVIEW

**Comment by caugner** [comment]
See also #1763 with the identical `__compat` data.

**Comment by Elchi3** [comment]
This API has sub features that should be added:
https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItemList/length
https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItemList/add
https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItemList/clear
https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItemList/DataTransferItem
https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItemList/remove

**Comment by caugner** [comment]
@Elchi I just added the missing sub features in 5fb67ff.

File: api/DataTransferItemList.json [hunk 1]
```diff
@@ -0,0 +1,293 @@
+{
+  "api": {
+    "DataTransferItem": {
```
Comment (connorshea) [inline api/DataTransferItemList.json]: This is using the wrong API name

**Pull request merged**